app/frontend/components/shared/form/text-form-control.tsx

```diff
@@ -0,0 +1,36 @@
+import { Flex, FormControl, FormControlProps, FormErrorMessage, FormLabel, Input, InputGroup } from "@chakra-ui/react"
+import React from "react"
+import { useFormContext } from "react-hook-form"
+import { useTranslation } from "react-i18next"
+
+interface ITextFormControlProps extends FormControlProps {
+  label: string
+  fieldName: string
+}
+
+export const TextFormControl = ({ label, fieldName }: ITextFormControlProps) => {
+  const { register, formState } = useFormContext()
+  const { t } = useTranslation()
+
+  return (
+    <FormControl isInvalid={!!formState?.errors[fieldName]} flex={1}>
+      <FormLabel>{label}</FormLabel>
```

Comment:
Probably needs for and id to connect to  correct input, unless FormControl takes care of it somehow?